File: ExamenMoria_Java/src/main/java/factories/personajesFactory/PersonajesFactory.java
package factories.personajesFactory;

import models.Personaje;
import models.item.Anillo;
import models.item.Carcaj;
import models.item.Vara;

import java.util.List;
import java.util.UUID;

import static utils.Utils.randomNumberGeneratorForIntegers;

public class PersonajesFactory {

    public static Personaje[] personajesGenerator(int poderMaximo, int flechasIniciales){
        Personaje[] personajes = new Personaje[3];

        personajes[0] = new Personaje(
                "Gandalf", true,
                List.of(
                        new Vara(UUID.randomUUID(), randomNumberGeneratorForIntegers(1, poderMaximo))
                )
        );

        personajes[1] = new Personaje(
                "Légolas", true,
                List.of(
                        new Carcaj(UUID.randomUUID(), randomNumberGeneratorForIntegers(1, flechasIniciales))
                )
        );

        personajes[2] = new Personaje(
                "Frodo", true,
                List.of(
                        new Anillo(UUID.randomUUID())
                )
        );

        return personajes;
    }
}

File: ExamenMoria-Python/projecto/factories/personajes/PersonajesFactory.py
import sys, os

path = os.path.abspath("projecto/models/personajes")

sys.path.append(path)

path = os.path.abspath("projecto/models/items")

sys.path.append(path)

import Vara, Carcaj, Anillo, Personaje, random

def createCaracters(poderMaximo, flechasIniciales) -> list:
    personajes = list()

    personajes.append(
        Personaje.Personaje("Gandalf", True, [Vara.Vara(1, random.randint(1, poderMaximo))])
    )

    personajes.append(
        Personaje.Personaje("Legolas", True, [Carcaj.Carcaj(1, random.randint(1, flechasIniciales))])
    )

    personajes.append(
        Personaje.Personaje("Frodo", True, [Anillo.Anillo(1, True)])
    )

    return personajes